lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Replicator.java

```diff
@@ -136,10 +136,10 @@ private static C4Replicator getReplicatorForHandle(long handle) {
 
     @Nullable
     private static C4Replicator getReplicatorForContext(@Nullable Object context) {
-        synchronized (CLASS_LOCK) { return CONTEXT_TO_C4_REPLICATOR_MAP.get(context); }
+        synchronized (CLASS_LOCK) { return (context == null) ? null : CONTEXT_TO_C4_REPLICATOR_MAP.get(context); }
     }
 
-    private static void addToMap(long handle, @NonNull C4Replicator repl, @NonNull Object context) {
+    private static void addToMap(long handle, @NonNull C4Replicator repl, @Nullable Object context) {
```

Comment:
If the context is null, I think we don't need to add the null key and replicator to CONTEXT_TO_C4_REPLICATOR_MAP.

Reply:
Good call.  Fixed.